app/experimenter/jetstream/client.py

```diff
@@ -261,7 +262,19 @@ def get_experiment_data(experiment):
                     errors_by_metric[metric_slug] = []
                 errors_by_metric[metric_slug].append(err)
             else:
-                errors_experiment_overall.append(err)
+                try:
+                    analysis_start_time = datetime.fromisoformat(
+                        experiment_metadata.get("analysis_start_time")
+                        if experiment_metadata is not None
+                        else ""
+                    )
+                    timestamp = datetime.fromisoformat(err.get("timestamp"))
+
+                    if timestamp >= analysis_start_time:
+                        errors_experiment_overall.append(err)
+                except (ValueError, TypeError):
+                    # ill-formatted/missing timestamp: default to including the error
+                    errors_experiment_overall.append(err)
 
```

Comment:
Oh I see what you mean, you just proceed as normal and if it throws then you just append the error.  Oh then yeah I wouldn't even bother with all the logic to do fallbacks, just look up the key and parse it and add `KeyError` to your exception handler.